rts/Lua/LuaSyncedCtrl.cpp

```diff
@@ -2243,6 +2243,10 @@ static bool SetSingleUnitWeaponState(lua_State* L, CWeapon* weapon, int index)
 			weapon->collisionFlags = lua_toint(L, index + 1);
 		} break;
 
+		case hashString("ttl"): {
+			weapon->ttl = (int) (lua_toint(L, index + 1) * GAME_SPEED);
```

Comment:
Should be float since now you can't pass e.g. 0.9 since it gets rounded to 0.
```suggestion
			weapon->ttl = (int) (lua_tonumber(L, index + 1) * GAME_SPEED);
```

Reply:
Whoops, my bad, thanks!